DotnetFoundation/DotnetFoundation.Api/Controllers/TaskDetailsController.cs

```diff
@@ -0,0 +1,147 @@
+using DotnetFoundation.Application.Models.DTOs.TaskDetailsDTO;
+using DotnetFoundation.Application.Models.Enums;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+
+namespace DotnetFoundation.Api.Controllers;
+
+[ApiController]
+[Route("api/tasks")]
+public class TaskDetailsController : ControllerBase
+{
+    private readonly ITaskDetailsService _TaskDetailsService;
+    public TaskDetailsController(ITaskDetailsService TaskDetailsService)
+    {
+        _TaskDetailsService = TaskDetailsService;
+    }
+
+    /// <summary>
+    /// Get all tasks.
+    /// </summary>
+    [HttpGet]
+    [Authorize]
+    [ProducesResponseType(StatusCodes.Status200OK)]
+    [ProducesResponseType(StatusCodes.Status500InternalServerError)]
+    public async Task<ActionResult<BaseResponse<List<TaskDetailsResponse>>>> GetAllTasksAsync()
+    {
+        BaseResponse<List<TaskDetailsResponse>> response = new(ResponseStatus.Fail);
+
+        response.Data = await _TaskDetailsService.GetAllTasksAsync().ConfigureAwait(false);
+        response.Status = ResponseStatus.Success;
+
+        return Ok(response);
+    }
+
+    /// <summary>
+    /// Get all active tasks.
+    /// </summary>
+    [HttpGet("active")]
+    [Authorize]
+    [ProducesResponseType(StatusCodes.Status200OK)]
+    [ProducesResponseType(StatusCodes.Status500InternalServerError)]
+    public async Task<ActionResult<BaseResponse<List<TaskDetailsResponse>>>> GetActiveTasksAsync()
+    {
+        BaseResponse<List<TaskDetailsResponse>> response = new(ResponseStatus.Fail);
+
+        response.Data = await _TaskDetailsService.GetActiveTasksAsync().ConfigureAwait(false);
+        response.Status = ResponseStatus.Success;
+
+        return Ok(response);
+    }
+
+    /// <summary>
+    /// Get task details by Id.
+    /// </summary>
+    /// <param name="taskId">Id of task record</param>
+    [HttpGet("{taskId}")]
+    [Authorize]
+    [ProducesResponseType(StatusCodes.Status200OK)]
+    [ProducesResponseType(StatusCodes.Status400BadRequest)]
+    [ProducesResponseType(StatusCodes.Status404NotFound)]
+    [ProducesResponseType(StatusCodes.Status500InternalServerError)]
+    public async Task<ActionResult<BaseResponse<TaskDetailsResponse>>> GetTaskByIdAsync(int taskId)
+    {
+        BaseResponse<TaskDetailsResponse> response = new(ResponseStatus.Fail);
+
+        response.Data = await _TaskDetailsService.GetTaskByIdAsync(taskId).ConfigureAwait(false);
+        response.Status = ResponseStatus.Success;
+
+        return Ok(response);
+    }
+
+    /// <summary>
+    /// Add new task.
+    /// </summary>
+    /// <param name="detailRequest">Role request details</param>
+    [HttpPost]
+    [Authorize]
+    [ProducesResponseType(StatusCodes.Status200OK)]
+    [ProducesResponseType(StatusCodes.Status400BadRequest)]
+    [ProducesResponseType(StatusCodes.Status500InternalServerError)]
+    public async Task<ActionResult<BaseResponse<string>>> InsertTaskAsync(TaskDetailsRequest detailRequest)
```

Comment:
The return type should be `BaseResponse<DTO>` instead of `BaseResponse<string>`

Reply:
The API should return the DTO object that is created and inserted in the database.

Reply:
Fixed